src/test/java/com/amihaiemil/docker/RtContainersTestCase.java

```diff
@@ -196,4 +196,100 @@ public void ioErrorIfResponseIs500() throws IOException {
             ), URI.create("http://localhost/test")
         ).create("some_image");
     }
+
+    /**
+     * Test for
+     * {@link RtContainers#create(String, String)}: The request URI should be
+     * welformed and include the 'name' query param.
+     * 
+     * @throws Exception unexpected
+     */
+    @Test
+    public void welformedUriForCreateNameImage() throws Exception {
+        new RtContainers(
+            new AssertRequest(
+                new Response(
+                    HttpStatus.SC_CREATED, "{ \"Id\": \"df2419f4\" }"
+                ),
+                new Condition(
+                    "Resource path must be /create?name=some_name",
+                    // @checkstyle LineLength (1 line)
+                    req -> req.getRequestLine().getUri().endsWith("/create?name=some_name")
+                )
+            ), URI.create("http://localhost/test")
+        ).create("some_name", "some_image");
+    }
+
+    /**
+     * Test for {@link RtContainers#create(JsonObject)}: request URI must be
+     * welformed and payload must include the input JSON.
+     * 
+     * @throws Exception unexpected
+     */
+    @Test
+    public void welformedUriAndPayloadForCreateJson() throws Exception {
+        final JsonObject json = Json.createObjectBuilder()
+            .add("Image", "ubuntu")
+            .add("Entrypoint", "script.sh")
+            .add("StopSignal", "SIGTERM")
+            .build();
+        new RtContainers(
+            new AssertRequest(
+                new Response(
+                    HttpStatus.SC_CREATED, "{ \"Id\": \"df2419f4\" }"
+                ),
+                new Condition(
+                    "Resource path must be /create",
+                    req -> req.getRequestLine().getUri().endsWith("/create")
+                ),
+                new Condition(
+                    "Payload must include the input JSON attributes.",
+                    req -> {
+                        final JsonObject payload = new PayloadOf(req);
+                        // @checkstyle LineLength (3 lines)
+                        return payload.getString("Image").equals(json.getString("Image"))
+                            && payload.getString("Entrypoint").equals(json.getString("Entrypoint"))
+                            && payload.getString("StopSignal").equals(json.getString("StopSignal"));
+                    }
+                )
+            ), URI.create("http://localhost/test")
+        ).create(json);
+    }
+
+    /**
+     * Test for {@link RtContainers#create(String, JsonObject)}: request URI
+     * must be  welformed and payload must include the input JSON.
+     * 
+     * @throws Exception unexpected
```

Comment:
@llorllale say ``@throws Exception If something goes wrong`` -- it's not clear what "unexpected" means. And the exception will indeed be thrown if anything goes wrong; in the lowest case it will be an ``AssertionError``, but it will still be thrown, that's how the test sfail.

Reply:
@amihaiemil ok